src/lib/pci_virtio_sock.c

```diff
@@ -1723,6 +1724,31 @@ static void apply_socket_config_set_buf_alloc(struct pci_vtsock_sock *s,
 	fprintf(stderr, "config: %s buf_alloc = %d\n", s->name, buf_alloc);
 }
 
+static void apply_socket_config_set_name(struct pci_vtsock_sock *s,
+					 struct vtsock_config_hdr *hdr)
+{
+	char *new_name;
+	char *msg_name = (char *)(hdr + 1);
+	size_t name_len = hdr->len - sizeof(struct vtsock_config_hdr);
+
+	name_len = strnlen(msg_name, name_len);
+
+	new_name = malloc(name_len + 22); /* max int string length + : + \0 */
```

Comment:
I'd put the : and \0 inside ' i.e. `+ ':' + '\0'` to mimic C character syntax and separate the characters from the + operators

I would also write `22` as `20 + 1 + 1` to make the relationship to the comment clearer. .

Then that would make me ask for a *#*define for the number 20, wherever that comes from.

Which finally leads me to ask why `s->name` then isn't a static inline array, since it is small and that avoids faffing around with dynamic allocation for this.